pkg/disk/obd.go

```diff
@@ -0,0 +1,182 @@
+/*
+ * MinIO Cloud Storage, (C) 2020 MinIO, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package disk
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"time"
+
+	"github.com/montanaflynn/stats"
+)
+
+const kb = uint64(1 << 10)
+const mb = uint64(kb << 10)
+const gb = uint64(mb << 10)
+
+var globalLatency = map[string]Latency{}
+var globalThroughput = map[string]Throughput{}
+
+// Latency holds latency information for write operations to the drive
+type Latency struct {
+	Avg          float64 `json:"avg_secs,omitempty"`
+	Percentile50 float64 `json:"percentile50_secs,omitempty"`
+	Percentile90 float64 `json:"percentile90_secs,omitempty"`
+	Percentile99 float64 `json:"percentile99_secs,omitempty"`
+	Min          float64 `json:"min_secs,omitempty"`
+	Max          float64 `json:"max_secs,omitempty"`
+}
+
+// Throughput holds throughput information for write operations to the drive
+type Throughput struct {
+	Avg          float64 `json:"avg_bytes/s,omitempty"`
```

Comment:
```suggestion
	Avg          float64 `json:"avg_bps,omitempty"`
```

Pretty sure slashes in a JSON name is a bad idea.

Reply:
it was bps before. The confusion arose between bits/sec and bytes/sec

Reply:
@wlan0 `avg_bytes_per_sec` or something like it then. No `/` pls.